graph-query-ir/src/main/java/oracle/pgql/lang/ir/QueryExpression.java

```diff
@@ -1188,8 +1190,65 @@ public void accept(QueryExpressionVisitor v) {
         v.visit(this);
       }
     }
+
+    class GetLatitude implements Function {
+
+      private final QueryExpression exp;
+
+      public GetLatitude(QueryExpression exps) {
+        this.exp = exps;
+      }
+
+      public QueryExpression getExp() {
+        return exp;
+      }
+
+      @Override
+      public ExpressionType getExpType() {
+        return ExpressionType.GET_LATITUDE;
+      }
+
+      @Override
+      public String toString() {
+        return "GET_LATITUDE(" + exp + ")";
+      }
+
+      @Override
+      public void accept(QueryExpressionVisitor v) {
+        v.visit(this);
+      }
+    }
+
+    class GetLongitude implements Function {
+
+      private final QueryExpression exp;
+
+      public GetLongitude(QueryExpression exps) {
+        this.exp = exps;
+      }
+
+      public QueryExpression getExp() {
+        return exp;
+      }
+
+      @Override
+      public ExpressionType getExpType() {
+        return ExpressionType.GET_LONGITUDE;
+      }
+
+      @Override
+      public String toString() {
+        return "GET_LONGITUDE(" + exp + ")";
+      }
+
+      @Override
+      public void accept(QueryExpressionVisitor v) {
+        v.visit(this);
+      }
+    }
```

Comment:
Hi Wilco, looks good but maybe we can put all the spatial functions in a separate class SpatialFunctions.java
I should've actually created separate files for every class in the first place but can't change that anymore.